cpp/src/arrow/compute/exec/exec_plan.cc

```diff
@@ -601,5 +618,215 @@ AsyncGenerator<util::optional<ExecBatch>> MakeSinkNode(ExecNode* input,
   return out;
 }
 
+std::shared_ptr<RecordBatchReader> MakeSinkNodeReader(ExecNode* input,
```

Comment:
Very minor thought, but having utility Iterator<RB> + schema => RBR and Generator<RB> + schema => RBR could be useful and then it would be more clear that the main goal here is ExecBatch -> RecordBatch.

Reply:
I'll extract those helpers